File: benchmarks/poly/fibonacci/fibonacci.php
<?php

function fib($n) {
    if ($n < 2) return $n;
    return fib($n - 1) + fib($n - 2);
}

function main() {
    $_t0 = hrtime(true);
    for ($i = 0; $i < 10; $i++) {
        echo fib($i) . "\n";
    }
    fwrite(STDERR, "bench_ns=" . (hrtime(true) - $_t0) . "\n");
}

main();

File: benchmarks/poly/fibonacci/fibonacci.lua
function fib(n)
  if n < 2 then return n end
  return fib(n - 1) + fib(n - 2)
end

function main()
  local _t0 = os.clock()
  for i = 0, 9 do
    print(fib(i))
  end
  io.stderr:write(string.format("bench_ns=%.0f\n", (os.clock() - _t0) * 1e9))
end

main()
